File: backend/app/schemas/department.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class DepartmentBase(BaseModel):
    name: str = Field(..., min_length=2, max_length=150)
    description: str | None = Field(None, max_length=500)


class DepartmentCreate(DepartmentBase):
    pass


class DepartmentUpdate(BaseModel):
    name: str | None = Field(None, min_length=2, max_length=150)
    description: str | None = Field(None, max_length=500)


class DepartmentResponse(DepartmentBase):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    company_id: uuid.UUID
    is_active: bool
    created_at: datetime
    updated_at: datetime


class PaginatedDepartmentResponse(BaseModel):
    items: list[DepartmentResponse]
    page: int
    page_size: int
    total: int
    pages: int

File: frontend/src/types/department.ts
export interface Department {
  id: string;
  name: string;
  description: string | null;
  is_active: boolean;
  company_id: string;
  created_at: string;
  updated_at: string;
}

export interface DepartmentCreatePayload {
  name: string;
  description?: string | null;
}

export interface DepartmentUpdatePayload {
  name?: string;
  description?: string | null;
}
